tools/code-owners-parser/CodeOwnersParser/CodeownersEntry.cs

```diff
@@ -106,11 +106,69 @@ public void ParseOwnersAndPath(string line)
             line = ParsePath(line);
             line = RemoveCommentIfAny(line);
 
-            foreach (string author in SplitLine(line, OwnerSeparator).ToList())
+            // If the line doesn't contain the OwnerSeparator AKA no owners, then the foreach loop below
+            // won't work. For example, the following line would end up causing "/sdk/communication" to
+            // be added as an owner when one is not listed
+            // /sdk/communication/
+            if (line.Contains(OwnerSeparator))
             {
-                if (!string.IsNullOrWhiteSpace(author))
-                    Owners.Add(author.Trim());
+                foreach (string author in SplitLine(line, OwnerSeparator).ToList())
+                {
+                    if (!string.IsNullOrWhiteSpace(author))
```

Comment:
To aid readability, I would extract the body of this `foreach` to a method with intention-revealing name.